File: typescript/src/shapes/rectangle.ts
import { Shape } from "./shape"

interface Rectangle extends Shape{
    width: number,
    height: number,
}

function newRectangle(width: number, height: number): Rectangle {
    return {
        width,
        height,
        area: function (): number {
            return width * height
        }
    }
}

export { Rectangle, newRectangle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Rectangle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Rectangle implements Shape {
    private double height;
    private double width;
    
    public Rectangle(double height, double width){
        this.height = height;
        this.width = width;
    }

    public double area() {
        return height * width;
    }

    public void draw(){
        System.out.println("Shape printed\n" + "Its area is " + area());
    }
}
